wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/graph/LinkedGraphMaker.scala

```diff
@@ -52,10 +52,15 @@ object LinkedGraphMaker {
         graph.findCycle match {
           case Some(cycle) =>
             val edgeStrings = cycle.value.edges map { case graph.EdgeT(from, to) => s""""${nodeName(from)}" -> "${nodeName(to)}"""" }
+            // we need to sort the edges lexicographically to make results deterministic. This helps testing.
+            val edges = edgeStrings.toVector.sorted
```

Comment:
The downside of sorting is that we lose the property that the strings are in cycle-order. Eg it's trivial to spot the cycle here:
```
b -> d
d -> c
c -> a
a -> b
```

vs the sorted version:
```
a -> b
b -> d
c -> a
d -> c
```

I wonder whether we can preserve both determinism and cycle order by always starting with the first name alphabetically, but then using the determined cycle order from then onwards?

In the above example that would mean:

```
a -> b
b -> d
d -> c
c -> a
```